tests/unit/test_client.py

```diff
@@ -104,6 +107,23 @@ def to_bytes_iterator(content, chunk_size=64 * 1024, exception_to_raise=None):
             raise exception_to_raise
 
 
+def mock_response(expected_range, blob_length, content, exception=None, etag=blob_etag):
+    response = mock.Mock(spec=StreamDownloadGenerator)
+    response.response = mock.Mock()
+    response.response.headers = {
+        "Content-Range": f"bytes {expected_range}/{blob_length}",
+        "ETag": etag,
+    }
+    response.iter_content_func = iter(
```

Comment:
Is there a particular reason we need to wrap the iterator from `to_bytes_iterator` with another iterator from `iter()`?